runtime/compiler/codegen/J9AheadOfTimeCompile.cpp

```diff
@@ -324,6 +325,68 @@ J9::AheadOfTimeCompile::initializeCommonAOTRelocationHeader(TR::IteratedExternal
          }
          break;
 
+      case TR_InlinedStaticMethodWithNopGuard:
+      case TR_InlinedSpecialMethodWithNopGuard:
+      case TR_InlinedVirtualMethodWithNopGuard:
+      case TR_InlinedInterfaceMethodWithNopGuard:
+         {
+         TR_RelocationRecordNopGuard *inlinedMethod = reinterpret_cast<TR_RelocationRecordNopGuard *>(reloRecord);
+         uintptrj_t destinationAddress = reinterpret_cast<uintptrj_t>(relocation->getTargetAddress());
+         TR_VirtualGuard *guard = reinterpret_cast<TR_VirtualGuard *>(relocation->getTargetAddress2());
+
+         uint8_t flags = 0;
+         // Setup flags field with type of method that needs to be validated at relocation time
+         if (guard->getSymbolReference()->getSymbol()->getMethodSymbol()->isStatic())
```

Comment:
These flags look kinda bogus to me (another example of using identify as a proxy to answer a question about a property), especially in conjunction with the set of records you're targeting in this PR, but any cleanup here can be deferred as it doesn't make sense for that to be in this PR.

Reply:
Those flags are actually used (under the SVM) when performing inlined site validations:

https://github.com/eclipse/openj9/blob/432df8c642770f89a95f2043d271434331faaafc/runtime/compiler/runtime/RelocationRecord.cpp#L2303-L2310

If the query is not the right one, then I guess it's a matter of finding the right query, but we definitely need to know whether the method is virtual or not (I feel like I remember bad things happening if you called `virtualMethodIsOverridden` on a static/special method heh).

Reply:
My point wasn't that the flags aren't used. My point was that the flags express a set of "identity"s used to compute a property that's only ever expressed as a combination of identities (`!isStatic && !isSpecial`). I think that's better expressed as the property itself rather than forcing everyone reading the code to figure out why those identities are the right ones for the code dependent on them.

Reply:
I see, yeah that makes sense.